src/us/kbase/workspace/database/provenance/Title.java

```diff
@@ -0,0 +1,171 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Optional;
+import us.kbase.workspace.database.Util;
+
+/**
+ * Represents a title or name of a resource.
+ */
+public class Title {
+
+	private final String titleString;
+	private final TitleType titleType;
+	private final String titleLanguage;
+
+	/**
+	 * An enum representing types of title.
+	 */
+	public enum TitleType {
```

Comment:
```suggestion
	public static enum TitleType {
```
No reason to bind an enum instance to its enclosing Title instance AFAICT

Reply:
:+1: